src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java

```diff
@@ -3186,4 +3188,81 @@ public void to_utc_timestamp() {
     Row[] expected = {Row.create(Timestamp.valueOf("2024-04-05 01:02:03.0"))};
     checkAnswer(df.select(Functions.to_utc_timestamp(df.col("a"))), expected, false);
   }
+
+  @Test
+  public void json_tuple1() {
+    DataFrame df =
+        getSession()
+            .sql(
+                "select parse_json(column1) as v, column2 as k from values ('{\"a\": null}','a'), "
+                    + "('{\"a\": \"foo\"}','a'), ('{\"a\": \"foo\"}','b'), (null,'a')");
+    df.show();
+    DataFrame jsontupleDF =
+        df.select(
+            (Functions.json_tuple(Functions.col("v"), Functions.col("k")).toArray(new Column[0])));
+    Row[] expected = {
+      Row.create((Object) null),
+      Row.create("foo"),
+      Row.create((Object) null),
+      Row.create((Object) null)
+    };
+    checkAnswer(jsontupleDF, expected, false);
+  }
+
+  @Test
+  public void json_tuple2() {
+    DataFrame df =
+        getSession()
+            .sql(
+                "select parse_json(column1) as v,column2 as id,column3 as name"
+                    + " from values ( '{\"id\": 5,\"name\": \"Jose\", \"age\": 29}','id','name','age')");
+
+    DataFrame jsontupleDF =
+        df.select(
+            (Functions.json_tuple(Functions.col("v"), Functions.col("id"), Functions.col("name"))
+                .toArray(new Column[0])));
+    jsontupleDF.show();
+    Row[] expected = {Row.create(("5"), ("Jose"))};
+    checkAnswer(jsontupleDF, expected, false);
+  }
+
+  @Test
+  public void try_cast() {
```

Comment:
need more test cases. You need to test several combinarions

df = spark.createDataFrame(
     [(2, "123"), (5, "Bob"), (3, None)], ["age", "name"])
df.select(df.name.try_cast(LongType())).show()
+----+
|name|
+----+
| 123|
|NULL|
|NULL|
+----+
Structure types should be consider as well:

df.filter(F.array_contains(F.lit('Key'), F.col('KEY_UNSTRCUTURED_ARRAY ').cast('ARRAY(VARCHAR)')).show()
# or
df.filter(F.array_contains(F.lit('Key'), F.col('KEY_UNSTRCUTURED_ARRAY ').cast(T.ArraryType(T.StringType()))).show()